### PR TITLE
Respect the NETRC environment variable

### PR DESCRIPTION
**Pull request by tarmath**
Resurecting https://github.com/psf/requests/pull/4339 with incorporated review/feedback

Fixes https://github.com/psf/requests/issues/4318

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,6 +13,8 @@ requests.egg-info/
 *.pyc
 *.swp
 *.egg
+.eggs/
+.tox/
 env/
 .venv/
 .eggs/
```
Comment (sigmavirus24) [inline .gitignore]: Why are you duplicating `.eggs` here? And `.tox` below?

Comment (tarmath) [reply .gitignore]: ah jeez, let me fix that...

Comment (tarmath) [reply .gitignore]: hrm, I amended the commit and force-pushed on my fork, but this change is still showing up...

Comment (tarmath) [reply .gitignore]: ok here goes.

Comment (sigmavirus24) [reply .gitignore]: GitHub does that to avoid confusing folks with "I left feedback here! They force-pushed and now it's disappeared!"

**Review by sigmavirus24 — DISMISSED** [review]

File: requests/utils.py [hunk 2]
```diff
@@ -169,12 +169,15 @@ def super_len(o):
...
+
     try:
         from netrc import netrc, NetrcParseError
 
         netrc_path = None
 
-        for f in NETRC_FILES:
+        for f in netrc_locations:
             try:
                 loc = os.path.expanduser('~/{}'.format(f))
```
Comment (nateprewitt) [inline requests/utils.py]: This doesn't look quite right. Are all `NETRC` paths relative to the users home directory? I think we only want this formatting for the `NETRC_FILES`.

**Review by nateprewitt — COMMENTED** [review]
Yeah, I think I'm fine with adding support. Left a quick question about the current implementation.

File: requests/utils.py (hunk 2 above)
Comment (nateprewitt) [reply requests/utils.py]: Actually, it looks like the last PR has [similar feedback](https://github.com/psf/requests/pull/4339#discussion_r144723124). If we can get that or something similar incorporated, I think we're set.

Comment (tarmath) [reply requests/utils.py]: very true... how's this then?

Comment (tarmath) [reply requests/utils.py]: I can also do it exactly as suggested in previous PR... let me know

File: requests/utils.py [hunk 3]
```diff
@@ -35,7 +35,7 @@
 from .exceptions import (
     InvalidURL, InvalidHeader, FileModeWarning, UnrewindableBodyError)
 
-NETRC_FILES = ('.netrc', '_netrc')
+NETRC_FILES = ('~/.netrc', '~/_netrc')
```
Comment (nateprewitt) [inline requests/utils.py]: I'm a little hesitant to mutate this directly with how long it's been around. Let's keep the formatting inside `get_netrc_auth` if possible.

Comment (tarmath) [reply requests/utils.py]: ah makes total sense, wouldn't have thought of that... changing then!

Comment (tarmath) [reply requests/utils.py]: done.

**Review by sigmavirus24 — APPROVED** [review]

**Review by nateprewitt — APPROVED** [review]
Great, this looks good to me. Thanks @tarmath!

**Comment by rgevaert** [comment]
@tarmath thanks for this!

**Pull request merged**